diff -uprN evince-0.8.1.orig/configure.ac evince-0.8.1/configure.ac
--- evince-0.8.1.orig/configure.ac	2007-04-09 22:35:24.000000000 +0200
+++ evince-0.8.1/configure.ac	2007-06-24 08:03:06.000000000 +0200
@@ -388,7 +388,7 @@ if test "x$enable_ps" = "xyes" ; then
 	EVINCE_MIME_TYPES="$EVINCE_MIME_TYPES;application/postscript;application/x-bzpostscript;application/x-gzpostscript;image/x-eps;image/x-bzeps;image/x-gzeps"
 fi
 if test "x$enable_dvi" = "xyes"; then
-	EVINCE_MIME_TYPES="$EVINCE_MIME_TYPES;application/x-dvi"
+	EVINCE_MIME_TYPES="$EVINCE_MIME_TYPES;application/x-dvi;application/x-gzdvi"
 fi
 if test "x$enable_djvu" = "xyes"; then
 	EVINCE_MIME_TYPES="$EVINCE_MIME_TYPES;image/vnd.djvu"
diff -uprN evince-0.8.1.orig/libdocument/ev-document-factory.c evince-0.8.1/libdocument/ev-document-factory.c
--- evince-0.8.1.orig/libdocument/ev-document-factory.c	2007-04-09 21:39:59.000000000 +0200
+++ evince-0.8.1/libdocument/ev-document-factory.c	2007-06-24 08:07:25.000000000 +0200
@@ -99,6 +99,7 @@ const EvDocumentType document_types[] = 
 #ifdef ENABLE_DVI
 	/* dvi: */
 	{"application/x-dvi",          EV_COMPRESSION_NONE, EV_BACKEND_DVI,  dvi_document_get_type},
+	{"application/x-gzdvi",        EV_COMPRESSION_GZIP, EV_BACKEND_DVI,  dvi_document_get_type},
 #endif
 
 #ifdef ENABLE_COMICS
