Description: Enable thumbnailing for gzipped/bzipped dvi files
Bug: http://bugzilla.gnome.org/show_bug.cgi?id=588013

diff -ru evince-2.22.0.orig/thumbnailer/evince-thumbnailer-dvi.schemas.in evince-2.22.0/thumbnailer/evince-thumbnailer-dvi.schemas.in
--- evince-2.22.0.orig/thumbnailer/evince-thumbnailer-dvi.schemas.in	2008-03-10 19:57:42.000000000 +0100
+++ evince-2.22.0/thumbnailer/evince-thumbnailer-dvi.schemas.in	2008-03-14 15:45:23.000000000 +0100
@@ -26,5 +26,58 @@
             </locale>
         </schema>
 
+
+        <schema>
+            <key>/schemas/desktop/gnome/thumbnailers/application@x-gzdvi/enable</key>
+            <applyto>/desktop/gnome/thumbnailers/application@x-gzdvi/enable</applyto>
+            <owner>evince</owner>
+            <type>bool</type>
+            <default>true</default>
+            <locale name="C">
+                <short></short>
+                <long></long>
+            </locale>
+        </schema>
+
+
+        <schema>
+            <key>/schemas/desktop/gnome/thumbnailers/application@x-gzdvi/command</key>
+            <applyto>/desktop/gnome/thumbnailers/application@x-gzdvi/command</applyto>
+            <owner>evince</owner>
+            <type>string</type>
+            <default>evince-thumbnailer -s %s %u %o</default>
+            <locale name="C">
+                <short></short>
+                <long></long>
+            </locale>
+        </schema>
+
+
+        <schema>
+            <key>/schemas/desktop/gnome/thumbnailers/application@x-bzdvi/enable</key>
+            <applyto>/desktop/gnome/thumbnailers/application@x-bzdvi/enable</applyto>
+            <owner>evince</owner>
+            <type>bool</type>
+            <default>true</default>
+            <locale name="C">
+                <short></short>
+                <long></long>
+            </locale>
+        </schema>
+
+
+        <schema>
+            <key>/schemas/desktop/gnome/thumbnailers/application@x-bzdvi/command</key>
+            <applyto>/desktop/gnome/thumbnailers/application@x-bzdvi/command</applyto>
+            <owner>evince</owner>
+            <type>string</type>
+            <default>evince-thumbnailer -s %s %u %o</default>
+            <locale name="C">
+                <short></short>
+                <long></long>
+            </locale>
+        </schema>
+
+
     </schemalist>
 </gconfschemafile>
