diff -ru evince-2.22.0.orig/backend/dvi/dvidocument.evince-backend.in evince-2.22.0/backend/dvi/dvidocument.evince-backend.in
--- evince-2.22.0.orig/backend/dvi/dvidocument.evince-backend.in	2008-03-10 19:57:31.000000000 +0100
+++ evince-2.22.0/backend/dvi/dvidocument.evince-backend.in	2008-03-14 15:41:14.000000000 +0100
@@ -1,4 +1,4 @@
 [Evince Backend]
 Module=dvidocument
 _TypeDescription=DVI Documents
-MimeType=application/x-dvi
+MimeType=application/x-dvi;application/x-gzdvi;application/x-bzdvi
diff -ru evince-2.22.0.orig/configure.ac evince-2.22.0/configure.ac
--- evince-2.22.0.orig/configure.ac	2008-03-10 19:58:27.000000000 +0100
+++ evince-2.22.0/configure.ac	2008-03-14 15:39:22.000000000 +0100
@@ -402,7 +402,7 @@
 	EVINCE_MIME_TYPES="${EVINCE_MIME_TYPES}application/postscript;application/x-bzpostscript;application/x-gzpostscript;image/x-eps;image/x-bzeps;image/x-gzeps;"
 fi
 if test "x$enable_dvi" = "xyes"; then
-	EVINCE_MIME_TYPES="${EVINCE_MIME_TYPES}application/x-dvi;"
+	EVINCE_MIME_TYPES="${EVINCE_MIME_TYPES}application/x-dvi;application/x-gzdvi;application/x-bzdvi;"
 fi
 if test "x$enable_djvu" = "xyes"; then
 	EVINCE_MIME_TYPES="${EVINCE_MIME_TYPES}image/vnd.djvu;"
diff -ru evince-2.22.0.orig/thumbnailer/evince-thumbnailer-dvi.schemas.in evince-2.22.0/thumbnailer/evince-thumbnailer-dvi.schemas.in
--- evince-2.22.0.orig/thumbnailer/evince-thumbnailer-dvi.schemas.in	2008-03-10 19:57:42.000000000 +0100
+++ evince-2.22.0/thumbnailer/evince-thumbnailer-dvi.schemas.in	2008-03-14 15:45:23.000000000 +0100
@@ -26,5 +26,58 @@
             </locale>
         </schema>
 
+
+        <schema>
+            <key>/schemas/desktop/gnome/thumbnailers/application@x-gzdvi/enable</key>
+            <applyto>/desktop/gnome/thumbnailers/application@x-gzdvi/enable</applyto>
+            <owner>evince</owner>
+            <type>bool</type>
+            <default>true</default>
+            <locale name="C">
+                <short></short>
+                <long></long>
+            </locale>
+        </schema>
+
+
+        <schema>
+            <key>/schemas/desktop/gnome/thumbnailers/application@x-gzdvi/command</key>
+            <applyto>/desktop/gnome/thumbnailers/application@x-gzdvi/command</applyto>
+            <owner>evince</owner>
+            <type>string</type>
+            <default>evince-thumbnailer -s %s %u %o</default>
+            <locale name="C">
+                <short></short>
+                <long></long>
+            </locale>
+        </schema>
+
+
+        <schema>
+            <key>/schemas/desktop/gnome/thumbnailers/application@x-bzdvi/enable</key>
+            <applyto>/desktop/gnome/thumbnailers/application@x-bzdvi/enable</applyto>
+            <owner>evince</owner>
+            <type>bool</type>
+            <default>true</default>
+            <locale name="C">
+                <short></short>
+                <long></long>
+            </locale>
+        </schema>
+
+
+        <schema>
+            <key>/schemas/desktop/gnome/thumbnailers/application@x-bzdvi/command</key>
+            <applyto>/desktop/gnome/thumbnailers/application@x-bzdvi/command</applyto>
+            <owner>evince</owner>
+            <type>string</type>
+            <default>evince-thumbnailer -s %s %u %o</default>
+            <locale name="C">
+                <short></short>
+                <long></long>
+            </locale>
+        </schema>
+
+
     </schemalist>
 </gconfschemafile>
