From: Jeremy Bicha <jbicha@ubuntu.com>
Date: Sun, 19 Aug 2018 18:02:53 -0400
Subject: shell: Update default app icon name

---
 shell/main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/shell/main.c b/shell/main.c
index 0901ff7..6c54078 100644
--- a/shell/main.c
+++ b/shell/main.c
@@ -296,7 +296,7 @@ main (int argc, char *argv[])
 
 	/* Manually set name and icon */
 	g_set_application_name (_("Document Viewer"));
-	gtk_window_set_default_icon_name ("evince");
+	gtk_window_set_default_icon_name ("org.gnome.Evince");
 
         application = ev_application_new ();
         if (!g_application_register (G_APPLICATION (application), NULL, &error)) {
