From: Jeremy Bicha <jbicha@ubuntu.com>
Date: Tue, 14 Aug 2018 10:28:52 -0400
Subject: shell: Update app icon name in about dialog

---
 shell/ev-application.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/shell/ev-application.c b/shell/ev-application.c
index 485045a..60d8a73 100644
--- a/shell/ev-application.c
+++ b/shell/ev-application.c
@@ -993,7 +993,7 @@ app_about_cb (GSimpleAction *action,
                                "authors", authors,
                                "documenters", documenters,
                                "translator-credits", _("translator-credits"),
-                               "logo-icon-name", "evince",
+                               "logo-icon-name", "org.gnome.Evince",
                                NULL);
 }
 
